packages/2d/src/components/Polygon.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  // PossibleSpacing,
+  BBox,
+  // SpacingSignal,
+} from '@motion-canvas/core/lib/types';
+import {Shape, ShapeProps} from './Shape';
+import {initial, signal} from '../decorators';
+import {SignalValue, SimpleSignal} from '@motion-canvas/core/lib/signals';
+
+function drawPolygon(
+  path: CanvasRenderingContext2D | Path2D,
+  rect: BBox,
+  // width: number,
+  // height: number,
+  sides: number,
+) {
+  const width = rect.width / 2;
+  const height = rect.height / 2;
+  for (let i = 0; i <= sides; i += 1) {
+    const theta = (i * 2 * Math.PI) / sides;
+    const x = width * Math.sin(theta);
+    const y = -height * Math.cos(theta);
+    if (i == 0) {
+      path.moveTo(x, y);
+    } else {
+      path.lineTo(x, y);
+    }
+  }
+  path.closePath();
+}
+
+export interface PolygonProps extends ShapeProps {
+  /**
+   * {@inheritDoc Polygon.sides}
+   */
+  sides?: SignalValue<number>;
+}
+
+export class Polygon extends Shape {
```

Comment:
Can we also add a short description for the node with an example or two?
Something like the `Circle` node:
https://github.com/motion-canvas/motion-canvas/blob/1c259d0d574bb56dbc8bc448300d9b94ee4d0bc4/packages/2d/src/components/Circle.ts#L20-L63

Reply:
Working on this.  Thanks for pointing me to the location where `Circle` is described.  This is helpful.

Reply:
Done